tests.yaml

```
@@ -570,8 +570,41 @@ tests:
 
 - interval: 1m
   input_series:
+  # node=minikube is uncordoned so we expect the alert to fire
   - series: 'kube_node_status_condition{condition="Ready",endpoint="https-main",cluster="kubernetes",instance="10.0.2.15:10250",job="kube-state-metrics",namespace="monitoring",node="minikube",pod="kube-state-metrics-b894d84cc-d6htw",service="kube-state-metrics",status="true"}'
+    values: '0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0'
+  - series: 'kube_node_spec_unschedulable{endpoint="https-main",cluster="kubernetes",instance="10.0.2.15:10250",job="kube-state-metrics",namespace="monitoring",node="minikube",pod="kube-state-metrics-b894d84cc-d6htw",service="kube-state-metrics",status="true"}'
+    values: '0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0'
+  # node=minikube2 is cordoned so we expect the alert to not fire
+  - series: 'kube_node_status_condition{condition="Ready",endpoint="https-main",cluster="kubernetes",instance="10.0.2.15:10250",job="kube-state-metrics",namespace="monitoring",node="minikube2",pod="kube-state-metrics-b894d84cc-f5e9f",service="kube-state-metrics"}'
+    values: '0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0'
+  - series: 'kube_node_spec_unschedulable{endpoint="https-main",cluster="kubernetes",instance="10.0.2.15:10250",job="kube-state-metrics",namespace="monitoring",node="minikube2",pod="kube-state-metrics-b894d84cc-f5e9f",service="kube-state-metrics"}'
+    values: '1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1'
+  alert_rule_test:
+  - eval_time: 18m
+    alertname: KubeNodeNotReady
+    exp_alerts:
+    - exp_labels:
+        cluster: kubernetes
+        node: minikube
+        severity: warning
```

Test failed in CI here.

As no aggregations are used in the `KubeNodeNotReady` alert expression, you must expect all labels from the original series. To resolve, ensure the expected labels match the series defined in the test, for example:

```suggestion
    - exp_labels:
        cluster: kubernetes
        node: minikube
        severity: warning
        condition: Ready
        endpoint: https-main
        instance: 10.0.2.15:10250
        job: kube-state-metrics
        namespace: monitoring
        pod: kube-state-metrics-b894d84cc-d6htw
        service: kube-state-metrics
        status: "true"
```